lib/addons/helm-addon/kubectl-provider.ts

```diff
@@ -0,0 +1,105 @@
+import { KubernetesManifest } from "@aws-cdk/aws-eks";
+import { Construct } from "@aws-cdk/core";
+import { ClusterInfo, Values } from "../..";
+
+
+export interface HelmChartConfiguration {
+    
+    /**
+     * Name of the helm chart (add-on)
+     */
+    name: string,
+    
+    /**
+     * Namespace where helm release will be installed
+     */
+    namespace: string,
+
+    /**
+     * Chart name
+     */
+    chart: string,
+
+    /**
+     * Helm chart version.
+     */
+    version: string, 
+
+    /**
+     * Helm release
+     */
+    release: string,
+
+    /**
+     * Helm repository
+     */
+    repository: string,
+
+    /**
+     * Optional values for the helm chart. 
+     */
+    values?: Values
+}
+
+type HelmChartDeployment = Required<HelmChartConfiguration>;
+
+export interface ManifestConfiguration {
+    name: string, 
+    namespace: string, 
+    manifestUrl: string
+}
+
+export interface ManifestDeployment extends Omit<ManifestConfiguration, "manifestUrl"> {
+    manifest: any,
+    values: Values
+}
+
+export class KubectlProvider {
+
+    constructor(private readonly clusterInfo : ClusterInfo) {}
+
+    public static helmProvider = function(clusterInfo: ClusterInfo, props: HelmChartDeployment) : Construct {
```

Comment:
consider renaming function to match CDK function name. I.E. `addHelmChart`.

Reply:
addressed